2002 uta nicholas refactor form validation

sfjsss opened the pull request:
-refactor back-end validation using JSR303
-serialize BindingResult using HashMap
-refactor form submission related controller methods to return ResponseEntity

nldlr committed:
Refactored controller methods (addUser, updateUser, addCar, updateCar). Address validation missing.

nldlr committed:
Refactored controller methods (addUser, updateUser, addCar, updateCar). Address validation missing.

chrandle committed:
user_id removed from car model, car object added to user model, data.sql script updated for new models, repositories being updated for new models

chrandle committed:
user_id removed from car model, car object added to user model, data.sql script updated for new models, repositories being updated for new models

chrandle committed:
removed columns from transient user members, added car association to data.sql

chrandle committed:
removed columns from transient user members, added car association to data.sql

nldlr committed:
Corrected code to make sure service runs only if input is valid. Also modified relevant comments.

nldlr committed:
Corrected code to make sure service runs only if input is valid. Also modified relevant comments.

chrisrchar committed:
removing comments and remaking hashcode, equals, tostring

chrisrchar committed:
fixed an error in sql script when rebasing to dev

chrisrchar committed:
Merge branch '2002-uta-ChristianRandle-User-Refactor' of https://github.com/Batch-747-Carolyn-Rehm/rideshare-user-service into 2002-uta-ChristianRandle-User-Refactor

chrisrchar committed:
removed unused imports

sfjsss committed:
resolve conflicts in UserController and CarController

sfjsss committed:
add logic in UserValidator to prevent potential nullPointerException

sfjsss committed:
add logic in UserValidator to prevent potential nullPointerException

chrisrchar committed:
Merge pull request #5 from Batch-747-Carolyn-Rehm/2002-uta-ChristianRandle-User-Refactor

User/Car Refactor

sfjsss committed:
Merge branch '2002-uta-nicholas-refactorFormValidation' of https://github.com/Batch-747-Carolyn-Rehm/rideshare-user-service into 2002-uta-nicholas-refactorFormValidation

carolynrehm commented on src/main/java/com/revature/beans/User.java:
```diff
@@ -199,6 +212,74 @@ public User(int userId, @NotBlank String userName, Batch batch, @NotBlank String
 		this.wState = wState;
 	}
+	
+	public User(@Valid @NotBlank @Size(min = 3, max = 12) @Pattern(regexp = "^\\w+\\.?\\w+$") String userName,
+			Batch batch,
+			@Valid @NotBlank @Size(max = 30) @Pattern(regexp = "^[a-zA-Z\\u00C0-\\u017F]+[- ]?[a-zA-Z\\u00C0-\\u017F]+$") String firstName,
+			@Valid @NotBlank @Size(max = 30) @Pattern(regexp = "^[a-zA-Z\\u00C0-\\u017F]+[- ]?[a-zA-Z\\u00C0-\\u017F]+$") String lastName,
+			@NotBlank @Email @Pattern(regexp = "^\\w+\\.?\\w+@\\w+\\.[a-zA-Z]{2,4}$") String email,
+			@NotBlank @Pattern(regexp = "^\\(?([0-9]{3})\\)?[-.\\s]?([0-9]{3})[-.\\s]?([0-9]{4})$") String phoneNumber,
+			boolean isDriver, boolean isActive, boolean isAcceptingRides, @NotBlank String hAddress,
+			@NotBlank String hCity, @NotBlank String hZip, @NotBlank String hState, @NotBlank String wAddress,
+			@NotBlank String wCity, @NotBlank String wZip, @NotBlank String wState, double distance, double duration,
+			Car car) {
```
There's a lot going on here, perhaps the builder pattern can be used instead? Also is the full work address necessary here? It seems like it will just be a selection of a few locations

sfjsss replied:
I suspect this constructor is useless so I commented it out and ran all features and they all worked. So I removed this constructor.

carolynrehm commented on src/main/java/com/revature/repositories/CarRepository.java:
```diff
@@ -24,6 +24,16 @@
 	 */
 	
-	@Query("select c from Car c where c.user.userId = ?1")
+	//@Query("select c from Car c where c.user.userId = ?1")
```
can this be removed?

chrisrchar replied:
I think this was from my pull request from earlier. We're working on a bug fix so we can remove that when we push later tonight.

sfjsss replied:
removed

sfjsss committed:
add cascadeType.All to car field in User Model so we can create a car as we update a user, remove address2 validation in address validator and add postal_code matching filter component

sfjsss committed:
remove commented out query in CarRepository

sfjsss committed:
remove unused constructor in User Class

sfjsss commented:
-add cascadeType.All to car field in User Model
-further optimize address validator and makes it more strict

sfjsss committed:
remove path variable on updateCar in CarController, remove validation logic on updateCar in CarServiceImpl since all validation is now handled by JSR303

sfjsss commented:
1. removed path variable on updateCar controller method since it was never used
2. removed the validation logic of updateCar inside of carServiceImpl since validation is now handled by JSR303

jaybouche reviewed (approved):
code changes work.
